src/app/auth/services/auth.service.ts

```diff
@@ -1,31 +1,59 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { User, UserRegistration } from '@shared/models/user.model';
+import { BehaviorSubject, map, Observable } from 'rxjs';
+import { SessionStorageService } from '@app/auth/services/session-storage.service';
+import { Router } from '@angular/router';
+
+interface LoginResponse {
+  result: string;
+}
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService {
-  login(user: any) {
-    // replace 'any' with the required interface
+  private API_URL = 'http://localhost:4000/';
+  private isAuthorized$$ = new BehaviorSubject(false);
+  public isAuthorized$ = this.isAuthorized$$.asObservable();
+
+  constructor(
+    private sessionService: SessionStorageService,
+    private http: HttpClient,
+    private router: Router
+  ) {
+    const token = this.sessionService.getToken();
+
+    this.isAuthorized$$.next(!!token);
+  }
+
+  login(user: User): Observable<void> {
     // Add your code here
+    return this.http.post<LoginResponse>(`${this.API_URL}login`, user).pipe(
```

Comment:
The URL paths (e.g. login, register, courses) can be added to a some constant or enum and reused. Avoid using magic strings.